public/assets/js/character_counter.js

```diff
@@ -0,0 +1,59 @@
+(function ($) {
```

Comment:
A lot of these look like third party plugins, I'd suggest separating these from your project JS files.  Possible create a plugins or vendor folder within the `public/assets/js` directory that houses your plugins.